packages/block-editor/src/components/use-no-recursive-renders/test/use-no-recursive-renders.js

```diff
@@ -4,29 +4,40 @@
 import { render } from '@testing-library/react';
 
 /**
- * WordPress dependencies
+ * Internal dependencies
  */
-import { Fragment } from '@wordpress/element';
-import { __experimentalUseNoRecursiveRenders as useNoRecursiveRenders } from '@wordpress/block-editor';
+import useNoRecursiveRenders from '../';
+import {
+	BlockEditContextProvider,
+	useBlockEditContext,
+} from '../../block-edit/context';
 
 // Mimics a block's Edit component, such as ReusableBlockEdit, which is capable
 // of calling itself depending on its `ref` attribute.
 function Edit( { attributes: { ref } } ) {
+	const { name } = useBlockEditContext();
```

Comment:
I understand the enhancement you made for taking into account block name and id, but these tests now are a bit confusing with the `ref` usage IMO. First thing I thought when seeing `ref` was React's ref, when the case is that we care about a unique id. A bit of renaming here might help probably, but I don't see this as blocking.

Reply:
Sure, we can use `uniqueId` to make it easier to follow. I think `ref` is coming from the Reusable blocks 😄

Reply:
Done it https://github.com/WordPress/gutenberg/pull/28456/commits/c0d579652e07395075328381269498001e031341.